sql/core/src/main/scala/org/apache/spark/sql/SparkSessionExtensions.scala

```diff
@@ -111,11 +112,12 @@ class SparkSessionExtensions {
   type FunctionDescription = (FunctionIdentifier, ExpressionInfo, FunctionBuilder)
   type TableFunctionDescription = (FunctionIdentifier, ExpressionInfo, TableFunctionBuilder)
   type ColumnarRuleBuilder = SparkSession => ColumnarRule
-  type QueryStagePrepRuleBuilder = SparkSession => Rule[SparkPlan]
+  type QueryStageRuleBuilder = SparkSession => Rule[SparkPlan]
 
   private[this] val columnarRuleBuilders = mutable.Buffer.empty[ColumnarRuleBuilder]
-  private[this] val queryStagePrepRuleBuilders = mutable.Buffer.empty[QueryStagePrepRuleBuilder]
+  private[this] val queryStagePrepRuleBuilders = mutable.Buffer.empty[QueryStageRuleBuilder]
   private[this] val runtimeOptimizerRules = mutable.Buffer.empty[RuleBuilder]
+  private[this] val queryStageOptimizerBuildersRules = mutable.Buffer.empty[QueryStageRuleBuilder]
```

Comment:
It's a little strange because of two plural words, `BuildersRules`. Is there any reason why we use ``queryStageOptimizerBuildersRules` instead of `queryStageOptimizerBuilders`?

Reply:
addressed